stablehlo/dialect/TypeInference.cpp

```diff
@@ -851,10 +852,21 @@ LogicalResult inferPadOp(
 
   auto inputShape = inputType.getShape();
   SmallVector<int64_t> resultShape;
+  ArrayRef<int64_t> inputBounds = encodingToBounds(inputType.getEncoding());
+  SmallVector<int64_t> resultBounds;
+
+  auto inferPadDim = [&](int64_t inputDim, int64_t paddingLowVal,
+                         int64_t paddingHighVal, int64_t paddingInteriorVal) {
+    return inputDim == ShapedType::kDynamicSize
+               ? ShapedType::kDynamicSize
+               : inputDim + paddingLowVal + paddingHighVal +
+                     std::max<int64_t>(inputDim - 1, 0LL) * paddingInteriorVal;
+  };
+
   for (int i = 0, e = inputShape.size(); i < e; i++) {
     if (hlo::isDynamicDimSize(inputShape[i])) {
```

Comment:
similar to the Slice op, move this logic after we check that paddingInteriorVal is not negative. That way, even if input dim is dynamic and bounds are missing, we report an error if paddingInteriorVal is negative.

Reply:
Sure, and the new code move verification in proper places: move as much up as possible.